homeassistant/components/dsmr_reader/sensor.py

```diff
@@ -0,0 +1,116 @@
+"""Support for DSMR Reader through MQTT."""
+import logging
+
+from homeassistant.components import mqtt
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import dt, slugify
+
+from .definitions import DEFINITIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "dsmr_reader"
+ATTR_UPDATED = "updated"
+ATTR_BEFORE_TRANSFORM = "before_transform"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up DSMR Reader sensors."""
+
+    sensors = []
+    for k in DEFINITIONS:
+        sensors.append(DSMRSensor(k))
+
+    async_add_entities(sensors)
+
+
+class DSMRSensor(Entity):
+    """Representation of a DSMR sensor that is updated via MQTT."""
+
+    def __init__(self, topic):
+        """Initialize the sensor."""
+
+        self._name = ""
+        self._entity_id = slugify(topic.replace("/", "_"))
+        self._friendly_name = ""
+        self._icon = ""
```

Comment:
I would like to suggest to make the code here shorter. E.g., 

```suggestion
        self._icon = self._definition["icon"] if "icon" in self._definition else None
```